openedx/features/user_leads/helpers.py

```diff
@@ -20,4 +20,7 @@ def save_user_utm(request):
     utm_params = get_utm_params(request)
 
     if not user.is_anonymous():
-        UserLeads.objects.get_or_create(user=user, origin=origin, **utm_params)
+        try:
+            UserLeads.objects.get(user=user, origin=origin)
+        except UserLeads.DoesNotExist:
+            UserLeads.objects.create(user=user, origin=origin, **utm_params)
```

Comment:
Your earlier approach was better then this, As that will query database only once and catch IntegrityError if already exist. Isn't it?

Reply:
Earlier approach was using `get_or_create` which essentially queries the database twice as well, once for get and once for create if nothing is returned in get. The issue is that we only want to `get(user, origin)` but `create(user, origin, **utm_params)`. That is not achievable with `get_or_create`. Putting an integrity error with `get_or_create` is the same as this approach, but uglier, because we do not need get_or_create just to catch an `IntegrityError`

Reply:
I'm talking about the approach you used before using get_or_create. 
```
        try:
            with transaction.atomic():
                UserLeads.objects.create(user=user, origin=origin, **utm_params)
        except IntegrityError:
            pass
```

Reply:
@saad-waqar and I agree that getting an `IntegrityError` on purpose and just `pass`ing it is not a good programming practice.

Reply:
I actually suggested this, I feel like relying on exceptions to achieve logic is a little weird/ugly.